core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BpmBillingProjectCreator.scala

```diff
@@ -60,15 +53,15 @@ class BpmBillingProjectCreator(billingRepository: BillingRepository,
    */
   override def postCreationSteps(createProjectRequest: CreateRawlsV2BillingProjectFullRequest,
                                  ctx: RawlsRequestContext
-  ): Future[Unit] =
+  ): Future[Unit] = {
+    val profileModel = blocking {
+      billingProfileManagerDAO.createBillingProfile(createProjectRequest.projectName.value,
+        createProjectRequest.billingInfo,
+        ctx
+      )
+    }
     for {
-      profileModel <- blocking {
-        billingProfileManagerDAO.createBillingProfile(createProjectRequest.projectName.value,
-                                                      createProjectRequest.billingInfo,
-                                                      ctx
-        )
-      }
       _ <- billingRepository.setBillingProfileId(createProjectRequest.projectName, profileModel.getId)
-    } yield {}
-
+    } yield { }
```

Comment:
Have the `yield` here because `setBillingProfileId` returns Future[Int]. We could change the return value for this method, but folks aren't using the return value and `validateBillingProjectCreationRequest` is also returning Future[Unit].